src/core/iomgr/socket_windows.c

```diff
@@ -64,8 +64,10 @@ void grpc_winsocket_shutdown(grpc_winsocket *socket) {
   shutdown_op(&socket->write_info);
 }
 
-void grpc_winsocket_orphan(grpc_winsocket *socket) {
-  grpc_iocp_socket_orphan(socket);
+void grpc_winsocket_orphan(grpc_winsocket *socket, int increase_g_orphans) {
+  if (increase_g_orphans) {
```

Comment:
This looks like this would leak the memory of the socket. Suggesting to:

1) rename increase_g_orphans to defer_destroy
2) add at the end of the function:

```
if (!defer_destroy) {
  grpc_winsocket_destroy(socket);
}